source/config/Config.cpp

```diff
@@ -433,6 +433,82 @@ bool PlainConfig::Validate() const
     return true;
 }
 
+void PlainConfig::SerializeToObject(Crt::JsonObject &object) const
+{
+    if (endpoint.has_value() && endpoint->c_str())
+    {
+        object.WithString(JSON_KEY_ENDPOINT, endpoint->c_str());
+    }
+    if (cert.has_value() && cert->c_str())
+    {
+        object.WithString(JSON_KEY_CERT, cert->c_str());
+    }
+    if (key.has_value() && key->c_str())
+    {
+        object.WithString(JSON_KEY_KEY, key->c_str());
+    }
+    if (rootCa.has_value() && rootCa->c_str())
+    {
+        object.WithString(JSON_KEY_ROOT_CA, rootCa->c_str());
+    }
+    if (thingName.has_value() && thingName->c_str())
+    {
+        object.WithString(JSON_KEY_THING_NAME, thingName->c_str());
+    }
+
+    JsonObject loggingObject;
+    logConfig.SerializeToObject(loggingObject);
+    object.WithObject(JSON_KEY_LOGGING, loggingObject);
+
+    JsonObject jobsObject;
+    jobs.SerializeToObject(jobsObject);
+    object.WithObject(JSON_KEY_JOBS, jobsObject);
+
+    JsonObject tunnelingObject;
+    tunneling.SerializeToObject(tunnelingObject);
+    object.WithObject(JSON_KEY_TUNNELING, tunnelingObject);
+
+    JsonObject deviceDefenderObject;
+    deviceDefender.SerializeToObject(deviceDefenderObject);
+    object.WithObject(JSON_KEY_DEVICE_DEFENDER, deviceDefenderObject);
+
+    JsonObject fleetProvisioningObject;
+    fleetProvisioning.SerializeToObject(fleetProvisioningObject);
+    object.WithObject(JSON_KEY_FLEET_PROVISIONING, fleetProvisioningObject);
+
+    if (fleetProvisioning.enabled)
+    {
+        JsonObject fleetProvisioningRuntimeObject;
+        fleetProvisioningRuntimeConfig.SerializeToObject(fleetProvisioningRuntimeObject);
+        object.WithObject(JSON_KEY_RUNTIME_CONFIG, fleetProvisioningRuntimeObject);
+    }
+
+    JsonObject samplesObject;
+    JsonObject pubSubObject;
+    pubSub.SerializeToObject(pubSubObject);
+    samplesObject.WithObject(JSON_KEY_PUB_SUB, pubSubObject);
+    object.WithObject(JSON_KEY_SAMPLES, samplesObject);
+
+    JsonObject configShadowObject;
+    configShadow.SerializeToObject(configShadowObject);
+    object.WithObject(JSON_KEY_CONFIG_SHADOW, configShadowObject);
+
+    JsonObject sampleShadowObject;
+    sampleShadow.SampleShadow::SerializeToObject(sampleShadowObject);
+    object.WithObject(JSON_KEY_SAMPLE_SHADOW, sampleShadowObject);
+
+    JsonObject secureElementObject;
+    secureElement.SerializeToObject(secureElementObject);
+    object.WithObject(JSON_KEY_SECURE_ELEMENT, secureElementObject);
+
+    if (sensorPublish.enabled)
```

Comment:
Same as above, should we remove this check?

Reply:
I want to keep this one since otherwise they have to define an empty array of sensors. But this is easily removed and the tests can be fixed.